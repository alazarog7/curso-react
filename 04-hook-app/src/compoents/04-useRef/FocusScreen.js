import React, { useRef } from 'react'

export const FocusScreen = () => {
    
    const intputRef = useRef();

    const handleClick = () => {
        intputRef.current.select();
    }

    return (
        <>
            <div className="mb-3">
                <input 
                    ref = {intputRef}
                    type="text" 
                    className="form-control" 
                    placeholder="Su Nombre"/>
            </div>
            <button className="btn btn-outline-primary" onClick={handleClick}>Focus</button>
        </>
    )
}
