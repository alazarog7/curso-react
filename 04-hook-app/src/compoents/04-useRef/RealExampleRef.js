import React, { useState } from 'react';
import { MultipleCustomHook } from '../03-example/MultipleCustomHook';

export const RealExampleRef = () => {

    const [show, setshow] = useState(false)
    return (
        <>
            <h1>Refexample</h1>
            <div style={{padding:"10px 0 10px 0"}}>
                {
                    show && <MultipleCustomHook /> 
                }
            </div>
            <button 
                className={`btn ${show?'btn-danger':'btn-success'}`}
                onClick={()=> {
                    setshow(!show)
                }}>
                {show?'OCULTAR':'MOSTRAR'}
            </button>
            
        </>
    )
}
