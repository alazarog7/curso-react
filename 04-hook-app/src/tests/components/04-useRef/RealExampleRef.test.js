import React from 'react';
import shallow from "enzyme/build/shallow"
import { RealExampleRef } from '../../../compoents/04-useRef/RealExampleRef';

describe('<RealExampleRef/>', () => {
    test('should show component correctly', () => {
        const wrapper = shallow(<RealExampleRef/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHook').exists()).toBe(false);
    })

    test('should show RealExampleRef', ()=>{
        const wrapper = shallow(<RealExampleRef/>)
        const button = wrapper.find('button');
        button.simulate('click');
        expect(wrapper.find('MultipleCustomHook').exists()).toBe(true);
    })
    
})
