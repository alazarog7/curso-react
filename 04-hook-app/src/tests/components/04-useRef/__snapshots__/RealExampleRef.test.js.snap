// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealExampleRef/> should show component correctly 1`] = `
<Fragment>
  <h1>
    Refexample
  </h1>
  <div
    style={
      Object {
        "padding": "10px 0 10px 0",
      }
    }
  />
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    MOSTRAR
  </button>
</Fragment>
`;
